addons/binding/org.openhab.binding.yamahareceiver/src/main/java/org/openhab/binding/yamahareceiver/internal/protocol/xml/InputConverterXML.java

```diff
@@ -0,0 +1,167 @@
+            for (String entry : entries) {
+                String[] keyValue = entry.split("="); // split the KEY=VALUE string
+                if (keyValue.length != 2) {
+                    logger.warn("Invalid setting: {} entry: {} - KEY=VALUE format was expected", setting, entry);
+                } else {
+                    String key = keyValue[0];
+                    String value = keyValue[1];
+
+                    if (map.putIfAbsent(key, value) != null) {
+                        logger.warn("Invalid setting: {} entry: {} - key: {} was already provided before", setting, entry, key);
+                    }
+                }
+            }
+        }
+        return map;
+    }
+
+    private Set<String> createInputsWithoutMapping() throws IOException, ReceivedMessageParseException {
+        // Tested on RX-S601D, RX-V479
+        Set<String> inputsWithoutMapping = Stream
+                .of(INPUT_SPOTIFY, INPUT_BLUETOOTH)
+                .collect(toSet());
+
+        Set<String> nativeInputNames = XMLProtocolService.getInputs(comReference.get(), Zone.Main_Zone)
+                .stream()
+                .filter(x -> x.isWritable())
+                .map(x -> x.getParam())
+                .collect(toSet());
+
+        // When native input returned matches any of 'HDMIx', 'AUDIOx' or 'NET RADIO', ensure no conversion happens.
+        // Tested on RX-S601D, RX-V479
+        nativeInputNames.stream()
+                .filter(x -> startsWithAndLength(x, "HDMI", 1)
+                        || startsWithAndLength(x, "AUDIO", 1)
+                        || x.equals(INPUT_NET_RADIO)
+                        || x.equals(INPUT_MUSIC_CAST_LINK))
+                .forEach(x -> inputsWithoutMapping.add(x));
+
+        return inputsWithoutMapping;
+    }
+
+    private static boolean startsWithAndLength(String str, String prefix, int extraLength) {
+        // Should be faster then regex
+        return str != null && str.length() == prefix.length() + extraLength && str.startsWith(prefix);
+    }
+
+    @Override
+    public String toCommandName(String name) {
+        // Note: conversation of outgoing command might be needed in the future
+        logger.trace("Converting from {} to command name {}", name, name);
+        return name;
+   }
+
+    @Override
+    public String fromStateName(String name) {
+
+        String convertedName;
+        String method;
+
+        if (inputMap.containsKey(name)) {
+            // Step 1: Check if the user defined custom mapping for their AVR
+            convertedName = inputMap.get(name);
+            method = "user defined mapping";
+        } else if (inputsWithoutMapping.contains(name)) {
+            // Step 2: Check if input should not be mapped at all
+            convertedName = name;
+            method = "no conversion rule";
+        } else {
+            // Step 3: Fallback to legacy logic
+            convertedName = convertNameToID(name);
+            method = "legacy mapping";
```

Comment:
The idea is that we remove this at some point, when enough testers report no "legacy mapping"s, I guess?